DataDog/datadog-lambda-python
# Detect exceptions raised outside of the handler

### What does this PR do?

This PR adds telemetry for exceptions that occur outside of the handler function. 

When an exception occurs outside of the handler function:
1. `aws.lambda.enhanced.errors` is emitted
2. A trace with an error span is created

### Motivation

- https://datadoghq.atlassian.net/browse/SVLS-4666
- Users should have visibility into this category of errors.
- #210 

### Testing Guidelines
1. Created a handler file like this:
```python
  import json
 
  0/0 # Purposely raise an exception

  def hello(event, context):
      body = {
          "message": "some message",
          "input": event,
      }
      return {"statusCode": 200, "body": json.dumps(body)}

```
2. Deployed my function and invoked it via function URL
3. Saw traces with an error span from the call ([example trace](https://ddserverless.datadoghq.com/apm/trace/15758361503635128623?colorBy=service&env=dev&graphType=flamegraph&shouldShowLegend=true&sort=time&spanID=5267772117214879748&spanViewType=errors&timeHint=1713881382599.7651))
4. Saw `aws.lambda.enhanced.errors` emitted ([graph](https://ddserverless.datadoghq.com/s/61ba5v0dnxcpd8dm/f47-xyc-kw4))
5. Traces appear in serverless detail page invocation table ([link to search for function](https://ddserverless.datadoghq.com/functions?search=functionname%3A%2Atal-dev%2A%20region%3Asa-east-1&cloud=aws&entity_view=lambda_functions&fromUser=false&panel_end=1713817896171&panel_paused=false&panel_start=1713731496171&panel_tab=errorTracking&shouldShowLegend=true&sort=time&spanViewType=logs&view=spans&start=1713731134137&end=1713817534137&paused=false))
6. Issues appear in error tracking and are informative

### Additional Notes



### Types of Changes


### Check all that apply


https://datadoghq.atlassian.net/browse/SVLS-4666
